### PR TITLE
fix(whitelist): Whitelist logic reformulated again

### PR DESCRIPTION
**Pull request by n4ch04**
### Context 

 Current whitelisting logic did not detect whitelisted element if  includes POSIX character classes 

### Description

Reformulate whitelisting logic to detect POSIX character classes  into whitelisted element

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: include/outputs [hunk 1]
```diff
@@ -169,16 +169,17 @@ textFail(){
   colorcode="$BAD"
   while read -r excluded_item; do
     ignore_check_name="${excluded_item%%:*}" # Check name is everything up to the first :
-    # Resource value is the second field of line included in whitelist divided by :
-    resource_value=$(awk -F ":" '{print $2}' <<< $excluded_item)
+    # Resource value is what it comes after CHECK_NAME: :
+    resource_value=$(awk -F "$CHECK_NAME:" '{print $2}' <<< "${excluded_item}")
```
Comment (ghost) [inline include/outputs:174]: Just a suggestion, use the same method as "ignore_check_name" above? ie `resource_value=${excluded_item#*:}`.

File: include/outputs [hunk 2]
```diff
@@ -169,16 +169,17 @@ textFail(){
...
     checked_value=$1
     if [[ "${ignore_check_name}" != "${CHECK_NAME}" ]]; then
       # not for this check
       continue
     fi
-    # To set WARNING flag both values must be exactly the same
-    if [[ "${checked_value}" == *"${resource_value}"* ]]; then
+    # To set WARNING flag checked_value have to include value of resource_value
+    # If it is treated as only expanse (*[]*) will not detect regex like [:alpha:]
+    if [[ "${checked_value}" =~ ${resource_value} ]]; then
```
Comment (ghost) [inline include/outputs:183]: This look good! Thanks for sorting it out :)

**Review by ghost — DISMISSED** [review]

**Review by MrCloudSec — APPROVED** [review]
Thank you @wobblesprout

File: include/outputs (hunk 1 above)
Comment (n4ch04) [reply include/outputs:174]: Thanks for your suggestion ! It's the same coded in a different way
We have scheduled the reformulation of whitelisting logic, including this, we just have changed ignore_check_name to follow the same syntax as resource_value, please check it again and let us know if works like it should

File: include/outputs (hunk 2 above)
Comment (n4ch04) [reply include/outputs:183]: You're welcome !!

File: include/outputs (hunk 1 above)
Comment (wobblesprout) [reply include/outputs:174]: Setting the delimiter to the name of the check in the awk command feel iffy to me - but if it works ... meh.

Personally I prefer using native bash when you can and not relying on external commands if you get the same result. While no one writes bash scripts with performance and resource usage being their main focus, it should at least be a consideration. Extra fork/exec/etc can start to add up, especially when we are running on a CI/CD pipeline that shares resources with other jobs. But again, this is only a suggestion - feel free to ignore.

**Review by jfagoagas — APPROVED** [review]
Good job @n4ch04 

🔝

**Pull request merged**